metals/src/main/scala/scala/meta/internal/metals/NewFilesProvider.scala

```diff
@@ -7,51 +7,120 @@ import MetalsEnrichments._
 import java.nio.file.Files
 import scala.concurrent.ExecutionContext
 import scala.util.control.NonFatal
+import org.eclipse.lsp4j.MessageType
+import org.eclipse.lsp4j.ExecuteCommandParams
+import org.eclipse.lsp4j.Location
+import DialogEnrichments._
+import scala.meta.internal.metals.Messages.NewScalaFile
 
 class NewFilesProvider(
     workspace: AbsolutePath,
-    packageProvider: PackageProvider
+    client: MetalsLanguageClient,
+    packageProvider: PackageProvider,
+    serverConfig: MetalsServerConfig,
+    focusedDocument: => Option[AbsolutePath]
 ) {
 
-  def createNewFile(directory: Option[URI], name: String, kind: String)(
+  private val classPick = MetalsPickItem(id = "class", label = "Class")
+  private val objectPick = MetalsPickItem(id = "object", label = "Object")
+  private val traitPick = MetalsPickItem(id = "trait", label = "Trait")
+  private val packageObjectPick =
+    MetalsPickItem(id = "package-object", label = "Package Object")
+  private val worksheetPick =
+    MetalsPickItem(id = "worksheet", label = "Worksheet")
+
+  def createNewFileDialog(directoryUri: Option[URI])(
       implicit ec: ExecutionContext
-  ): Future[URI] = kind match {
-    case "class" | "object" | "trait" =>
-      createClass(directory, name, kind)
-    case "package-object" =>
-      createPackageObject(directory)
-    case "worksheet" =>
-      createWorksheet(directory, name)
-    case invalid => Future.failed(new IllegalArgumentException(invalid))
+  ): Future[Unit] = {
+    val directory = directoryUri
+      .map(_.toString.toAbsolutePath)
+      .orElse(focusedDocument.map(_.parent))
+
+    val newlyCreatedFile =
+      askForKind
```

Comment:
I've made some kind of DSL here, to cover `Future[Option[_]]` handling (making this method more or readable). `Future` is for about remote requests, and `Option` reflect that the request can be cancelled (thus, nothing should be returned eventually).
Not sure, if it's good here.